### PR TITLE
feat(otel upgrade/new components): Upgraded otel to v0.50.0 and Added Components

### PR DESCRIPTION
**Pull request by cpheps**
### Proposed Change
- Updated OTel dependencies to v0.50.0
- Updated googlecloudexporter and hardcoded to legacy config for now
- Added the following receivers:
  - Varnish
  - Riak
  - Microsoft SQL Server
  - Microsoft IIS
  - Active Directory Domain
  - vCenter

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Review by BinaryFissionGames — APPROVED** [review]

File: docs/installation-mac.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ sh -c "$(curl -fsSlL https://github.com/observiq/observiq-otel-collector/release
 
 ### Additional Install Steps
 
-If you plan on JMX collecting metrics via the [JMX Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.49.0/receiver/jmxreceiver/README.md) you can copy the `opentelemetry-java-contrib-jmx-metrics.jar` to the default location to make configuration easier.
+If you plan on JMX collecting metrics via the [JMX Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.50.0/receiver/jmxreceiver/README.md) you can copy the `opentelemetry-java-contrib-jmx-metrics.jar` to the default location to make configuration easier.
```
Comment (StefanKurek) [inline docs/installation-mac.md:29]: nit and won't hold up PR for this but would be better if it was like "If you plan on collecting metrics via that [JMX Receiver]..."

File: collector/settings.go [hunk 2]
```diff
@@ -19,27 +19,40 @@ import (
...
 	factories, _ := factories.DefaultFactories()
 	buildInfo := component.BuildInfo{
 		Command:     os.Args[0],
 		Description: buildDescription,
 		Version:     version,
 	}
-	provider := service.MustNewDefaultConfigProvider(configPaths, []string{})
 
-	return service.CollectorSettings{
+	fmp := filemapprovider.New()
```
Comment (StefanKurek) [inline collector/settings.go:39]: @cpheps can you give a quick summary of what these changes are about?

File: cmd/collector/main.go [hunk 3]
```diff
@@ -36,8 +36,12 @@ func main() {
 	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 	defer cancel()
 
-	settings := collector.NewSettings(*configPaths, version.Version(), nil)
-	if err := run(ctx, settings); err != nil {
+	settings, err := collector.NewSettings(*configPaths, version.Version(), nil)
+	if err != nil {
+		log.Fatal(err)
```
[the remark this answers is not in the document]
Comment (cpheps) [reply cmd/collector/main.go:41]: The log.Fatal will do an `os.Exit(1)` after it logs its message. If I didn't address what you're saying let me know.

[the remark this answers is not in the document]
Comment (StefanKurek) [reply cmd/collector/main.go:41]: @cpheps nope you addressed it. good to know about log.Fatal

**Review by StefanKurek — APPROVED** [review]

**Pull request merged**